src/shared/lib/AnnotationUtils.ts

```diff
@@ -45,18 +50,113 @@ export function geneToMyCancerGenome(myCancerGenomes:IMyCancerGenome[]):IMyCance
-        map[key] = true;
+    hotspots.forEach((hotspot:HotspotMutation) => {
+        let geneIndex = index[hotspot.hugoSymbol.toUpperCase()];
+
+        if (!geneIndex)
+        {
+            geneIndex = {};
+
+            index[hotspot.hugoSymbol.toUpperCase()] = geneIndex;
+        }
+
+        const hotspotType = hotspot.type || defaultHotspotType || "hotspot";
+        let typeIndex = index[hotspot.hugoSymbol.toUpperCase()][hotspotType];
+
+        if (!typeIndex)
+        {
+            typeIndex = {
+                hotspotMutations: [],
+                hotspotSet: new HotspotSet([]) // this is a placeholder to avoid TS errors.
+                                               // overwritten by the initHotspotSets function.
+            };
+
+            index[hotspot.hugoSymbol.toUpperCase()][hotspotType] = typeIndex;
+        }
+
+        typeIndex.hotspotMutations.push(hotspot);
     });
 
-    return map;
+    initHotspotSets(index);
+
+    return index;
+}
+
+export function initHotspotSets(hotspotIndex: IHotspotIndex)
+{
+    _.keys(hotspotIndex).forEach((geneSymbol:string) => {
+        _.keys(hotspotIndex[geneSymbol]).forEach((hotspotType:string) => {
+            const intervals = hotspotIndex[geneSymbol][hotspotType].hotspotMutations.map(
+                (hotspot:HotspotMutation):[number, number] => {
+                    // start (and optionally end) positions
+                    const positions = hotspot.residue.match(/[0-9]+/g) || [];
+
+                    let start = -1;
+                    let end = -1;
+
+                    if (positions.length > 0) {
+                        start = parseInt(positions[0]);
+                        end = start;
+                    }
+
+                    // overwrite end if exists
+                    if (positions.length > 1) {
+                        end = parseInt(positions[1]);
+                    }
+
+                    return [start, end];
+                }
+            );
+
+            hotspotIndex[geneSymbol][hotspotType].hotspotSet = new HotspotSet(intervals);
+        });
+    });
+}
+
+export function isHotspot(mutation:Mutation, index:IHotspotIndex):boolean
+{
+    let isHotspot = false;
+    let hotspotType:string|null = null;
+
+    if (mutation.mutationType.toLowerCase().indexOf("missense") > -1) {
+        hotspotType = "single residue";
+    }
+    else if (mutation.mutationType.toLowerCase().indexOf("inframe") > -1 ||
+        mutation.mutationType.toLowerCase().indexOf("in_frame") > -1 ||
+        mutation.mutationType.toLowerCase().indexOf("in frame") > -1)
+    {
+        hotspotType = "in-frame indel";
+    }
+
+    if (hotspotType &&
+        index[mutation.gene.hugoGeneSymbol])
+    {
+        const indexedData = index[mutation.gene.hugoGeneSymbol][hotspotType];
+
+        if (indexedData)
+        {
+            isHotspot = indexedData.hotspotSet.check(mutation.proteinPosStart, mutation.proteinPosEnd);
```

Comment:
I think it is better to add more checks here, we may have `-1` or `0` instead of `undefined` as a `proteinPosEnd` value